### PR TITLE
Add hide / show layer methods

### PR DESCRIPTION
**Pull request by elenatorro**
Related with https://github.com/CartoDB/carto-vl/issues/757

## Checklist

* [x] Add show method
* [x] Add hide method
* [x] Disable interactivity when layer is hidden
* [x] Enable interactivity when layer is hidden
* [x] Add debug example
* [x] Add layer integration tests
* [x] Add layer tests
* [x] Add documentation

### CODE REVIEW

File: src/Layer.js [hunk 1]
```diff
@@ -281,22 +291,57 @@ export default class Layer {
...
+     */
     hasDataframes () {
         return this._renderLayer.hasDataframes();
     }
 
+    /**
+     *  Get layer id
+     *
+     * @memberof carto.Layer
+     * @instance
```
Comment (davidmanzanares) [inline src/Layer.js]: Documenting all of this and exposing it on the API has consequences, some of this methods are probably broken, they certainly are untested, users haven't requested them and some of them can be confusing. For example, `getSource` and `getViz` semantics are not clear, when an `update` is performed the source is not changed immediately. All of this should be documented if we decide to put it on the API. To be honest, I think documenting this is not a good idea at this moment

File: src/Layer.js [hunk 2]
```diff
@@ -305,8 +350,49 @@ export default class Layer {
...
+     * Change layer visibility to visible
+     *
+     * @memberof carto.Layer
+     * @instance
+     * @api
+     *
+     * @fires updated
+     */
+    show () {
+        this.visible = true;
```
Comment (davidmanzanares) [inline src/Layer.js]: If `visible` is part of the API, then it should have a setter, if not it should be `_visible`

File: src/Layer.js [hunk 3]
```diff
@@ -305,8 +350,49 @@ export default class Layer {
...
+        this._integrator.changeVisibility(this._id, this.visibility);
+        this._fire('updated', { visibility: this.visibility });
```
Comment (davidmanzanares) [inline src/Layer.js]: Do we really need to put the visibility in the event? this change requires to change the documentation and to fire with the same property in all points of the program. I would prefer to keep it simple stupid (KISS)

File: src/constants/layer.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const layerVisibility = Object.freeze({
+    VISIBLE: 'visible',
```
Comment (davidmanzanares) [inline src/constants/layer.js]: In my honest opinion this is less readable than putting `'visible'`

File: src/integrator/Map.js [hunk 5]
```diff
@@ -76,6 +80,37 @@ export default class Map {
...
+                this.show(layerId);
+                break;
+            case layerVisibility.HIDDEN:
+                this.hide(layerId);
+                break;
+        }
+    }
+
+    hide (layerId) {
+        for (let index = 0; index < this._layers.length; index++) {
```
Comment (davidmanzanares) [inline src/integrator/Map.js]: This and `show()` can be done with a `Set` with much less complexity

File: src/constants/layer.js (hunk 4 above)
Comment (elenatorro) [reply src/constants/layer.js]: In this case is not a matter of readability, but of reusability.

**Comment by elenatorro** [comment]
🚀

**Pull request merged**